src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -430,23 +413,42 @@ export class BrowsingContextProcessor {
       .catch((error) => this.#logger?.(LogType.debugError, error));
   }
 
+  #createCdpTarget(
+    targetCdpClient: CdpClient,
+    targetInfo: Protocol.Target.TargetInfo,
+    sessionId: string
+  ) {
+    this.#setEventListeners(targetCdpClient);
+
+    return CdpTarget.create(
+      targetInfo.targetId,
+      targetCdpClient,
+      this.#browserCdpClient,
+      sessionId,
+      this.#realmStorage,
+      this.#eventManager,
+      this.#preloadScriptStorage,
+      this.#networkStorage,
+      this.#acceptInsecureCerts,
+      this.#logger
+    );
+  }
+
   #workers = new Map<string, Realm>();
   #handleWorkerTarget(cdpTarget: CdpTarget, browsingContextId: string) {
     cdpTarget.cdpClient.on('Runtime.executionContextCreated', (params) => {
       const {uniqueId, id, origin} = params.context;
-      const realm = new Realm(
-        this.#realmStorage,
-        this.#browsingContextStorage,
-        uniqueId,
-        browsingContextId,
-        id,
-        serializeOrigin(origin),
-        'dedicated-worker',
-        undefined,
+      const realm = new DedicatedWorkerRealm(
         cdpTarget.cdpClient,
         this.#eventManager,
-        this.#sharedIdWithFrame,
-        this.#logger
+        id,
+        this.#logger,
+        serializeOrigin(origin),
+        this.#realmStorage.getRealm({
```

Comment:
What if the browsing context has JS sandboxes?

Reply:
In theory it's possible, but we currently don't support it :/ But you're right, it should be defaultRealm